diff --git a/kogito-springboot-examples/decisiontable-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/RestQueryTest.java b/kogito-springboot-examples/decisiontable-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/RestQueryTest.java
index 30860d23f..4e3121e9b 100644
--- a/kogito-springboot-examples/decisiontable-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/RestQueryTest.java
+++ b/kogito-springboot-examples/decisiontable-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/RestQueryTest.java
@@ -19,7 +19,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.queries.KogitoSpringbootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DashboardsListTest.java b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DashboardsListTest.java
index a0ae9e8eb..dccbf1c39 100644
--- a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DashboardsListTest.java
+++ b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DashboardsListTest.java
@@ -22,7 +22,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.autoconfigure.actuate.metrics.AutoConfigureMetrics;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
diff --git a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DroolsMetricsTest.java b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DroolsMetricsTest.java
index abebfb6da..b4b1f7870 100644
--- a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DroolsMetricsTest.java
+++ b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/DroolsMetricsTest.java
@@ -19,7 +19,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.autoconfigure.actuate.metrics.AutoConfigureMetrics;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/LoanEligibilityTest.java b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/LoanEligibilityTest.java
index 894f46759..ba697254c 100644
--- a/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/LoanEligibilityTest.java
+++ b/kogito-springboot-examples/dmn-drools-springboot-metrics/src/test/java/org/kie/kogito/examples/springboot/LoanEligibilityTest.java
@@ -19,7 +19,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.autoconfigure.actuate.metrics.AutoConfigureMetrics;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
diff --git a/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationListenerTest.java b/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationListenerTest.java
index a17dfd4c2..0239fcae0 100644
--- a/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationListenerTest.java
+++ b/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationListenerTest.java
@@ -27,7 +27,7 @@ import org.kie.kogito.decision.DecisionEventListenerConfig;
 import org.kie.kogito.dmn.springboot.example.mock.MockDMNRuntimeEventListener;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationTest.java b/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationTest.java
index d440b2f91..18dcf9eba 100644
--- a/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationTest.java
+++ b/kogito-springboot-examples/dmn-listener-springboot/src/test/java/org/kie/kogito/dmn/springboot/example/listener/TrafficViolationTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.springboot.example.listener;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNMiningModelTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNMiningModelTest.java
index 3d9de5cda..4dbc61d2f 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNMiningModelTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNMiningModelTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.pmml.springboot.example;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNRegressionTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNRegressionTest.java
index 0db95dc71..6fbda4ff5 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNRegressionTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNRegressionTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.pmml.springboot.example;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNScoreCardTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNScoreCardTest.java
index 6287eaa2a..4320fc756 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNScoreCardTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNScoreCardTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.pmml.springboot.example;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNTreeTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNTreeTest.java
index 3885a8b88..ee1064bb4 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNTreeTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DMNTreeTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.pmml.springboot.example;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DecisionTreeTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DecisionTreeTest.java
index 617b87735..d39fd3534 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DecisionTreeTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/DecisionTreeTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/LinRegTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/LinRegTest.java
index 24ff9ec0c..37a9d745b 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/LinRegTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/LinRegTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/MiningModelTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/MiningModelTest.java
index 0c57345b4..95452775f 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/MiningModelTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/MiningModelTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/ScoreCardTest.java b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/ScoreCardTest.java
index 64a10b51d..655408c6f 100644
--- a/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/ScoreCardTest.java
+++ b/kogito-springboot-examples/dmn-pmml-springboot-example/src/test/java/org/kie/kogito/dmn/pmml/springboot/example/ScoreCardTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-springboot-example/src/test/java/org/kie/kogito/dmn/springboot/example/TrafficViolationTest.java b/kogito-springboot-examples/dmn-springboot-example/src/test/java/org/kie/kogito/dmn/springboot/example/TrafficViolationTest.java
index cdd423732..f38894360 100644
--- a/kogito-springboot-examples/dmn-springboot-example/src/test/java/org/kie/kogito/dmn/springboot/example/TrafficViolationTest.java
+++ b/kogito-springboot-examples/dmn-springboot-example/src/test/java/org/kie/kogito/dmn/springboot/example/TrafficViolationTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.dmn.springboot.example;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/dmn-tracing-springboot/src/test/java/org/kie/kogito/dmn/springboot/tracing/LoanEligibilityIT.java b/kogito-springboot-examples/dmn-tracing-springboot/src/test/java/org/kie/kogito/dmn/springboot/tracing/LoanEligibilityIT.java
index e3db89f92..becc38fc4 100644
--- a/kogito-springboot-examples/dmn-tracing-springboot/src/test/java/org/kie/kogito/dmn/springboot/tracing/LoanEligibilityIT.java
+++ b/kogito-springboot-examples/dmn-tracing-springboot/src/test/java/org/kie/kogito/dmn/springboot/tracing/LoanEligibilityIT.java
@@ -28,7 +28,7 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/flexible-process-springboot/src/test/java/org/kie/kogito/flexible/example/springboot/ServiceDeskProcessTest.java b/kogito-springboot-examples/flexible-process-springboot/src/test/java/org/kie/kogito/flexible/example/springboot/ServiceDeskProcessTest.java
index cdcfba2ad..a0843eb7b 100644
--- a/kogito-springboot-examples/flexible-process-springboot/src/test/java/org/kie/kogito/flexible/example/springboot/ServiceDeskProcessTest.java
+++ b/kogito-springboot-examples/flexible-process-springboot/src/test/java/org/kie/kogito/flexible/example/springboot/ServiceDeskProcessTest.java
@@ -27,7 +27,7 @@ import org.kie.kogito.flexible.example.model.SupportCase;
 import org.kie.kogito.flexible.example.service.TriageService;
 import org.kie.kogito.tests.KogitoSpringbootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.annotation.DirtiesContext.ClassMode;
 
diff --git a/kogito-springboot-examples/onboarding-springboot/src/main/java/org/kie/kogito/examples/WorkItemHandlerConfig.java b/kogito-springboot-examples/onboarding-springboot/src/main/java/org/kie/kogito/examples/WorkItemHandlerConfig.java
index 674769d92..02c0f90a2 100644
--- a/kogito-springboot-examples/onboarding-springboot/src/main/java/org/kie/kogito/examples/WorkItemHandlerConfig.java
+++ b/kogito-springboot-examples/onboarding-springboot/src/main/java/org/kie/kogito/examples/WorkItemHandlerConfig.java
@@ -15,7 +15,6 @@
  */
 package org.kie.kogito.examples;
 
-import jakarta.annotation.PostConstruct;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
@@ -34,6 +33,8 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
+import jakarta.annotation.PostConstruct;
+
 @Component
 public class WorkItemHandlerConfig extends DefaultWorkItemHandlerConfig {
 
diff --git a/kogito-springboot-examples/onboarding-springboot/src/test/java/org/kie/kogito/examples/onboarding/OnboardingEndpointIT.java b/kogito-springboot-examples/onboarding-springboot/src/test/java/org/kie/kogito/examples/onboarding/OnboardingEndpointIT.java
index 03279f764..4a7dc01e2 100644
--- a/kogito-springboot-examples/onboarding-springboot/src/test/java/org/kie/kogito/examples/onboarding/OnboardingEndpointIT.java
+++ b/kogito-springboot-examples/onboarding-springboot/src/test/java/org/kie/kogito/examples/onboarding/OnboardingEndpointIT.java
@@ -34,7 +34,7 @@ import org.kie.kogito.process.ProcessConfig;
 import org.kie.kogito.testcontainers.springboot.InfinispanSpringBootTestResource;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
diff --git a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/LinRegTest.java b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/LinRegTest.java
index e2be1f17a..ea8ec3850 100644
--- a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/LinRegTest.java
+++ b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/LinRegTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/MiningModelTest.java b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/MiningModelTest.java
index 875c7a2f1..cf0d651c6 100644
--- a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/MiningModelTest.java
+++ b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/MiningModelTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SampleMineTest.java b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SampleMineTest.java
index 686593bf2..af14c7aec 100644
--- a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SampleMineTest.java
+++ b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SampleMineTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SimpleScorecardTest.java b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SimpleScorecardTest.java
index 0ab88633a..58852e3c4 100644
--- a/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SimpleScorecardTest.java
+++ b/kogito-springboot-examples/pmml-springboot-example/src/test/java/org/kie/kogito/pmml/springboot/example/SimpleScorecardTest.java
@@ -21,7 +21,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-decisions-rules-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java b/kogito-springboot-examples/process-decisions-rules-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
index 35fe70cf4..a28941480 100644
--- a/kogito-springboot-examples/process-decisions-rules-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
+++ b/kogito-springboot-examples/process-decisions-rules-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
@@ -22,7 +22,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
diff --git a/kogito-springboot-examples/process-decisions-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java b/kogito-springboot-examples/process-decisions-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
index b3e8048df..69c650592 100644
--- a/kogito-springboot-examples/process-decisions-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
+++ b/kogito-springboot-examples/process-decisions-springboot/src/test/java/org.kie.kogito.traffic/TrafficProcessIT.java
@@ -22,7 +22,7 @@ import java.util.Map;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
diff --git a/kogito-springboot-examples/process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java b/kogito-springboot-examples/process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
index 6495d06f8..79875699d 100644
--- a/kogito-springboot-examples/process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
+++ b/kogito-springboot-examples/process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
@@ -21,7 +21,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.testcontainers.springboot.InfinispanSpringBootTestResource;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-mongodb-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java b/kogito-springboot-examples/process-mongodb-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
index 66682d68d..0b88d67e7 100644
--- a/kogito-springboot-examples/process-mongodb-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
+++ b/kogito-springboot-examples/process-mongodb-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
@@ -20,7 +20,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.testcontainers.springboot.MongoDBSpringBootTestResource;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/DashboardGenerationTest.java b/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/DashboardGenerationTest.java
index abc62570f..a4b22b143 100644
--- a/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/DashboardGenerationTest.java
+++ b/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/DashboardGenerationTest.java
@@ -23,7 +23,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/ProcessMetricsTest.java b/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/ProcessMetricsTest.java
index 67e78bfa5..489d18b49 100644
--- a/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/ProcessMetricsTest.java
+++ b/kogito-springboot-examples/process-monitoring-springboot/src/test/java/org/kie/kogito/examples/springboot/ProcessMetricsTest.java
@@ -25,7 +25,7 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.autoconfigure.actuate.metrics.AutoConfigureMetrics;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-postgresql-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java b/kogito-springboot-examples/process-postgresql-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
index 2cd9176a6..83e8ecd7c 100644
--- a/kogito-springboot-examples/process-postgresql-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
+++ b/kogito-springboot-examples/process-postgresql-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java
@@ -20,7 +20,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.testcontainers.springboot.PostgreSqlSpringBootTestResource;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestIT.java b/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestIT.java
index 975926ef5..209a7245a 100644
--- a/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestIT.java
+++ b/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/OrdersRestIT.java
@@ -26,7 +26,7 @@ import org.kie.kogito.testcontainers.springboot.KafkaSpringBootTestResource;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
diff --git a/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/PersonsRestIT.java b/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/PersonsRestIT.java
index ca00f4a9b..b31ac9498 100644
--- a/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/PersonsRestIT.java
+++ b/kogito-springboot-examples/process-springboot-example/src/test/java/org/kie/kogito/examples/demo/PersonsRestIT.java
@@ -27,7 +27,7 @@ import org.kie.kogito.testcontainers.springboot.KafkaSpringBootTestResource;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
diff --git a/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot-with-console/src/test/java/org/acme/travels/ApprovalsRestIT.java b/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot-with-console/src/test/java/org/acme/travels/ApprovalsRestIT.java
index e45fefd32..b39fb4769 100644
--- a/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot-with-console/src/test/java/org/acme/travels/ApprovalsRestIT.java
+++ b/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot-with-console/src/test/java/org/acme/travels/ApprovalsRestIT.java
@@ -25,7 +25,7 @@ import org.kie.kogito.testcontainers.springboot.KafkaSpringBootTestResource;
 import org.kie.kogito.testcontainers.springboot.KeycloakSpringBootTestResource;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
diff --git a/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot/src/test/java/org/acme/travels/ApprovalsRestIT.java b/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot/src/test/java/org/acme/travels/ApprovalsRestIT.java
index 32f63e398..37d1a3bd9 100644
--- a/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot/src/test/java/org/acme/travels/ApprovalsRestIT.java
+++ b/kogito-springboot-examples/process-usertasks-with-security-oidc-springboot/src/test/java/org/acme/travels/ApprovalsRestIT.java
@@ -23,7 +23,7 @@ import org.kie.kogito.springboot.KogitoSpringbootApplication;
 import org.kie.kogito.testcontainers.springboot.KeycloakSpringBootTestResource;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
diff --git a/kogito-springboot-examples/process-usertasks-with-security-springboot/src/test/java/org/acme/travels/security/springboot/ApprovalsRestTest.java b/kogito-springboot-examples/process-usertasks-with-security-springboot/src/test/java/org/acme/travels/security/springboot/ApprovalsRestTest.java
index 18d52fdca..8ec6a8a98 100644
--- a/kogito-springboot-examples/process-usertasks-with-security-springboot/src/test/java/org/acme/travels/security/springboot/ApprovalsRestTest.java
+++ b/kogito-springboot-examples/process-usertasks-with-security-springboot/src/test/java/org/acme/travels/security/springboot/ApprovalsRestTest.java
@@ -20,7 +20,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.kie.kogito.tests.KogitoInfinispanSpringbootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.annotation.DirtiesContext.ClassMode;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
diff --git a/kogito-springboot-examples/rules-legacy-springboot-example/src/test/java/org/kie/kogito/RestQueryTest.java b/kogito-springboot-examples/rules-legacy-springboot-example/src/test/java/org/kie/kogito/RestQueryTest.java
index cd61a4ad9..17ea83ca0 100644
--- a/kogito-springboot-examples/rules-legacy-springboot-example/src/test/java/org/kie/kogito/RestQueryTest.java
+++ b/kogito-springboot-examples/rules-legacy-springboot-example/src/test/java/org/kie/kogito/RestQueryTest.java
@@ -19,7 +19,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.legacy.KogitoSpringbootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
diff --git a/kogito-springboot-examples/ruleunit-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/ruleunit/RestQueryTest.java b/kogito-springboot-examples/ruleunit-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/ruleunit/RestQueryTest.java
index 8f54299cf..9a94bc193 100644
--- a/kogito-springboot-examples/ruleunit-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/ruleunit/RestQueryTest.java
+++ b/kogito-springboot-examples/ruleunit-springboot-example/src/test/java/org/kie/kogito/decisiontable/springboot/ruleunit/RestQueryTest.java
@@ -19,7 +19,7 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.queries.KogitoSpringbootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 
 import io.restassured.RestAssured;
